csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -336,6 +336,52 @@ void inferenceAllocationOrder(
   }
 }
 
+namespace {
+// Propagate allocation orders from an SDPA's inputs to outputs. This is
+// necessary to make an SPDA's allocation domain consistent with the output
+// at::Tensor from expression evaluation. Currently, we call ATen to evaluate
+// SDPAs so matching their behavior, despite being fragile, is the best
+// solution.
+class SdpaPropagator : public OptOutConstDispatch {
```

Comment:
Users in nvfuser can set stride_order for these inputs `fd.add_output(out, stride_order)`.  In that case, we should not be propagating this order since it will lead to errors. It will likely require inserting tensor ops within these ops.

Not requesting a change in this PR specifically. This was part of the motivation for Issue #2425. Related issue for matmul/linear: Issue #2427.

Reply:
That's right -- I'll change `propagateAllocation` to only propagate when the destination doesn't have allocation.